app/sample/tx_video_split_sample.c

```diff
@@ -0,0 +1,260 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright(c) 2022 Intel Corporation
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <pthread.h>
+#include <signal.h>
+#include <st20_dpdk_api.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "../src/app_platform.h"
+
+#define TX_VIDEO_PMD ST_PMD_DPDK_USER
+#define TX_VIDEO_PORT_BDF "0000:4b:00.0"
+#define TX_VIDEO_YUV_FILE "../assets/test_4k.yuv"
+
+#define TX_VIDEO_UDP_PORT (20000)
+#define TX_VIDEO_PAYLOAD_TYPE (112)
+
+/* local ip address for current bdf port */
+static uint8_t g_tx_video_local_ip[ST_IP_ADDR_LEN] = {192, 168, 96, 1};
+/* dst ip address for tx video session */
+static uint8_t g_tx_video_dst_ip[ST_IP_ADDR_LEN] = {239, 168, 85, 20};
+
+static bool g_video_active = false;
+static st_handle g_st_handle;
+
+struct app_context {
+  int idx;
+  int fb_send;
+  int nfi; /* next_frame_idx */
+  st20_tx_handle handle;
+  struct st20_tx_ops ops;
+
+  size_t frame_size; /* 1080p */
+  size_t fb_size;    /* 4k */
+  int fb_idx;
+  int fb_total;
+  size_t fb_offset;
+
+  st_dma_mem_handle dma_mem;
+};
+
+static int tx_video_next_frame(void* priv, uint16_t* next_frame_idx,
+                               struct st20_tx_frame_meta* meta) {
+  struct app_context* s = priv;
+  int ret = 0;
+
+  if (!s->handle) return -EIO; /* not ready */
+
+  struct st20_ext_frame ext_frame;
+  ext_frame.buf_addr =
+      st_dma_mem_addr(s->dma_mem) + s->fb_idx * s->fb_size + s->fb_offset;
+  ext_frame.buf_iova =
+      st_dma_mem_iova(s->dma_mem) + s->fb_idx * s->fb_size + s->fb_offset;
+  ext_frame.buf_len = s->frame_size * 2;
+  st20_tx_set_ext_frame(s->handle, s->nfi, &ext_frame);
+
+  *next_frame_idx = s->nfi;
+  s->nfi++;
+  if (s->nfi >= 3) s->nfi = 0;
```

Comment:
Can we use add fb_cnt to the context, 3 is hardcode here. Assign fb_cnt in the init.